prow/cmd/pipeline/main.go

```diff
@@ -118,7 +118,7 @@ func main() {
 		logrus.WithError(err).Fatal("failed to load prow config")
 	}
 
-	configs, err := kube.LoadClusterConfigs(o.kubeconfig, "")
+	configs, err := kube.LoadClusterConfigs(o.kubeconfig, "", "", "")
```

Comment:
This just passed the threshold for ugly function signature, especially when the values are actually optional, and you need to pass empty strings around.

Please refactor this to something sensible, the options are:
- Parameter object (https://www.refactoring.com/catalog/introduceParameterObject.html)
- Fluent API like `kube.Config().FromFile(path).FromDirectory(path, suffix).FromTokenFile(path).Load()`
- [Functional Options](https://www.sohamkamani.com/golang/options-pattern/): `kube.Config(File(path), Dir(path, suffix))`

Reply:
I will try the 3rd one.

Reply:
Done.